code/modules/projectiles/guns/ballistic/launchers.dm

```diff
@@ -139,19 +139,33 @@
 /obj/item/gun/ballistic/rocketlauncher/update_icon()
 	icon_state = "[initial(icon_state)]-[chambered ? "1" : "0"]"
 
-/obj/item/gun/ballistic/rocketlauncher/suicide_act(mob/user)
+/obj/item/gun/ballistic/rocketlauncher/suicide_act(mob/living/user)
+	user.visible_message("<span class='warning'>[user] aims [src] at the ground! It looks like they're about to perform a sick rocket jump!<span>", \
+		"<span class='userdanger'>You aim [src] at the ground, ready to perform a bisnasty rocket jump...</span>")
 	if(can_shoot())
-		handle_suicide(user, user, null, TRUE)
+		if(chambered && chambered.BB)
```

Comment:
can_shoot already does this check

Reply:
should i delete it from the base proc as well?

Reply:
wouldn't myself just since can_shoot can change, this one's just a specific case where it's already covered.